src/history/hs_rec.c

```diff
@@ -609,6 +609,9 @@ __wt_hs_insert_updates(
                 WT_ASSERT(session, __txn_visible_id(session, upd->txnid));
 #endif
 
+            /* Clear out the insert success flag prior to our insert attempt. */
```

Comment:
Instead of this function to clear the insert success flag, how about clearing in the `__curhs_insert` function itself.

Reply:
If `__hs_insert_record` returns non-zero before reaching the `__curhs_insert` call then it may wrongly think that we inserted properly.

Reply:
Yes, understood.